test/1_process_engine/interprocess_communication_test.js

```diff
@@ -149,7 +155,77 @@ describe('Inter-process communication - ', () => {
       const evaluationCallback = (message) => {
         if (message.flowNodeId === endEventToWaitFor) {
           should(message).have.property('currentToken');
-          should(message.currentToken).be.match(expectedResult);
+          should(message.currentToken).be.eql(initialPayload);
+          resolve();
+        }
+      };
+
+      // Subscribe for the EndEvent
+      eventAggregator.subscribeOnce(endMessageToWaitFor, evaluationCallback);
+
+      // Run the process that is supposed to publish the signal.
+      testFixtureProvider.executeProcess(processModelIntermediateThrowEventTests, 'StartEvent_SendSignal', correlationId, initialPayload);
+    });
+  });
+
+  it('should correctly use a customized payload with a MessageThrowEvent', async () => {
+
+    const correlationId = uuid.v4();
+    const endEventToWaitFor = 'EndEvent_MessageReceived';
+
+    const initialPayload = {
+      mode: 'custom',
+      message: 'message triggered',
+    };
+
+    // We can't await the process execution here, because that would prevent us from starting the second process.
+    // As a result we must subscribe to the event that gets send when the test is done.
+    testFixtureProvider.executeProcess(processModelIntermediateCatchEventTests, 'StartEvent_MessageTest', correlationId);
+
+    await processInstanceHandler.waitForProcessInstanceToReachSuspendedTask(correlationId, processModelIntermediateCatchEventTests);
+
+    return new Promise((resolve) => {
+
+      const endMessageToWaitFor = `/processengine/correlation/${correlationId}/processmodel/${processModelIntermediateCatchEventTests}/ended`;
+      const evaluationCallback = (message) => {
+        if (message.flowNodeId === endEventToWaitFor) {
```

Comment:
Extract Business Rule

```suggestion
        const messageIsAboutEndEventToWaitFor: boolean = message.flowNodeId === endEventToWaitFor;
        if (messageIsAboutEndEventToWaitFor) {
```

(Maybe you can find a better name for this)